### PR TITLE
feat: reserve internal column

### LINKED ISSUE #2392: mito: reserves internal columns
We need to validate the request that it doesn't contain internal column names.

https://github.com/GreptimeTeam/greptimedb/blob/26992d58cd8ab67bcdc0b5c65a7f49d3cad80d7e/src/store-api/src/metadata.rs#L353-L356

Internal columns contain
https://github.com/GreptimeTeam/greptimedb/blob/26992d58cd8ab67bcdc0b5c65a7f49d3cad80d7e/src/store-api/src/storage/consts.rs#L78-L85

### PR DESCRIPTION
**Pull request by Nateiru**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?
The column name created by user cannot be the name of the reserved column.

Closes #2392

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [ ]  I have added the necessary unit tests and integration tests.

## Refer to a related PR or issue link (optional)
#2392

### CODE REVIEW

File: src/store-api/src/storage/consts.rs [hunk 1]
```diff
@@ -84,6 +84,17 @@ pub const OP_TYPE_COLUMN_NAME: &str = "__op_type";
...
 
+/// Internal Column Name 
+static INTERNAL_COLUMN_VEC: [&str; 3] = [
+    SEQUENCE_COLUMN_NAME,
+    OP_TYPE_COLUMN_NAME,
+    PRIMARY_KEY_COLUMN_NAME,
+];
+
+pub fn is_internal_column(name: &str) -> bool {
+    INTERNAL_COLUMN_VEC.iter().any(|&column| name.contains(column))
```
Comment (evenyag) [inline src/store-api/src/storage/consts.rs:95]: Does array provide `contains()` method?
```suggestion
    INTERNAL_COLUMN_VEC.contains(name)
```

File: src/store-api/src/metadata.rs [hunk 2]
```diff
@@ -368,6 +369,16 @@ impl RegionMetadata {
             );
         }
 
+        ensure!(
+            !is_internal_column(&column_metadata.column_schema.name),
+            InvalidMetaSnafu {
```
Comment (evenyag) [inline src/store-api/src/metadata.rs:374]: Please add a test case for this.

File: src/store-api/src/storage/consts.rs [hunk 3]
```diff
@@ -104,4 +115,14 @@ mod tests {
...
     }
+
+    #[test]
+    fn test_is_internal_column() {
+        assert!(is_internal_column(SEQUENCE_COLUMN_NAME));
+        assert!(is_internal_column(OP_TYPE_COLUMN_NAME));
+        assert!(is_internal_column(PRIMARY_KEY_COLUMN_NAME));
+
+        let valid_name: String = "GreptimeDB".to_string() + SEQUENCE_COLUMN_NAME;
+        assert!(!is_internal_column(&valid_name));
```
Comment (evenyag) [inline src/store-api/src/storage/consts.rs:126]:
```suggestion
        assert!(!is_internal_column("my_column"));
        assert!(!is_internal_column("my_sequence"));
```

**Review by evenyag — APPROVED** [review]

**Review by fengjiachun — APPROVED** [review]
LGTM

**Review by waynexia — APPROVED** [review]

**Pull request merged**